### PR TITLE
Jormun: Check that the origin and destination of journeys are not the same

### PR DESCRIPTION
**Pull request by AurelienLP**
https://jira.kisio.org/browse/NAVITIAII-2944

`if not (args['destination'] or args['origin'])` is check earlier because it can.

We check `if api == 'journeys' and args['origin'] == args['destination']` and send a message error.

The tests `test_no_origin_or_destination_point_in_request` and `test_same_origin_and_destination_point_in_request` are added.

The test `test_journeys_do_not_loose_precision` is change to use two different coords for `from` and `to`.

### CODE REVIEW

File: source/jormungandr/tests/journey_common_tests.py [hunk 1]
```diff
@@ -346,24 +346,6 @@ def test_journeys_date_valid_not_zeropadded(self):
         assert 'message' in response
         assert "Unable to parse datetime, year" in response['message']
 
-    def test_journeys_do_not_loose_precision(self):
```
Comment (benoit-bst) [inline source/jormungandr/tests/journey_common_tests.py:349]: why is it deleted ?
This test is used to check different precisions

File: source/jormungandr/tests/journey_common_tests.py [hunk 2]
```diff
@@ -346,24 +346,6 @@ def test_journeys_date_valid_not_zeropadded(self):
...
-        """do we have a good precision given back in the id"""
```
Comment (mbounabi) [inline source/jormungandr/tests/journey_common_tests.py:350]: Why was this test added when it was? Because it was clearly added for a purpose?

**Review by benoit-bst — APPROVED** [review]

**Comment by benoit-bst** [comment]
Jenkins, test this please !

**Comment by benoit-bst** [comment]
Jenkins, test this please !

**Comment by AurelienLP** [comment]
Jenkins, test this please.

**Comment by AurelienLP** [comment]
Jenkins, test this please.

**Review by pbougue — COMMENTED** [review]
We still have the issue if there is 1m of distance between from and to.
IMHO this is managing in code a specific case that should properly be managed in a more generic way. If we merge that, the next ticket we will probably get on that subject will be about the 1m journey, and when we will manage it, this code/behavior will have to be maintained for retrocompat (worst case) or cleaned.

So I think we should use this case as a way to improve many more and not "hide" it in between (unless there is an urgent need on that).

**Comment by AurelienLP** [comment]
Jenkins, test this please.

**Review by mbounabi — APPROVED** [review]

**Comment by AurelienLP** [comment]
Jenkins, test this please.

**Review by xlqian — APPROVED** [review]

**Pull request merged**